### ISSUE TITLE
Issue with latest_block_based_cache_middleware

### ISSUE THREAD

**Issue by Meemaw** [ethereum/web3.py#2185]
* Version: 5.24.0
* Python: 3.9.5
* OS: osx

### What was wrong?

Please include any of the following that are applicable:

There seems to be an issue with the `latest_block_based_cache_middleware` implementation. Given the following config:

```python
w3 = Web3(HTTPProvider(settings.ETHEREUM_NODE_URL, request_kwargs={"timeout": 8}))

# Set the gas price strategy
# https://web3py.readthedocs.io/en/stable/gas_price.html#selecting-the-gas-price-strategy
# NOTE - It's recommended to add a cache solution because of how inneficient this gas strategy is
w3.eth.setGasPriceStrategy(fast_gas_price_strategy)

w3.middleware_onion.add(middleware.time_based_cache_middleware)
w3.middleware_onion.add(middleware.latest_block_based_cache_middleware)
w3.middleware_onion.add(middleware.simple_cache_middleware)
```

I'm observing `w3.eth.get_block('latest')` to always be cached even though it should never be cached.

Digging into the code a bit more, I suspect the issue in in `_is_latest_block_number_request` fn, https://github.com/ethereum/web3.py/blob/489394726d2718325ce6ff1d3ff57764b9495086/web3/middleware/cache.py#L317, specifically the part where params are compared: https://github.com/ethereum/web3.py/blob/489394726d2718325ce6ff1d3ff57764b9495086/web3/middleware/cache.py#L320

I've printed params locally, and it seems that `params[:1]` evaluates to `('latest',)` (tuple) and thus the comparison with list fails.

### How can it be fixed?

Convert params slice to list before comparing?

**Comment by kclowes** [comment]
Thanks for raising this! This does look like a bug, will fix!

**Comment by B3nac** [comment]
Dibs! If nobody is working on this already. :smile:

**Comment by pacrob** [comment]
Fixed in #2346
